.travis.yml

```diff
@@ -36,7 +36,10 @@ matrix:
         - KARMEN_CLOUD_MODE=0 docker-compose up --build -d || travis_terminate 1;
         - while ! curl -I localhost:4000 > /dev/null 2>&1; do echo 'Waiting for server...'; sleep 1; done
         - npm run test:cypress
-
+        - docker-compose down
+        - KARMEN_CLOUD_MODE=1 docker-compose up --build -d
+        - while ! curl -I localhost:4000 > /dev/null 2>&1; do echo 'Waiting for server...'; sleep 1; done
+        - npm run test:cypress
```

Comment:
Why do you run it again? It's a wrong merge isn't it?

Reply:
We want to run the tests in both modes - cloud and non cloud.

Reply:
It would be really nice if we could only run the non-cloud tests when running again to save precious build time secs. This for example looks like a really neat way of doing so:

https://stackoverflow.com/a/56298024/303184

Reply:
I don't agree because the backend is also dependent on mode/env var, so we would run all tests again.

Reply:
I don't understand, we're talking about E2E tests here, don't we? How are backend tests involved in this?

Reply:
However we do E2E tests, on the backend side can be a different flow of processing  dependent on the mode, thus I suppose we would run all tests for both modes. Your suggestion is applicable if we would test against a API mock.

Regarding to the link to stackoverflow, we cannot use the Cypress enviroment variable to determine which tests should be skipped. In this case of E2E tests, we read the mode from the window object of the application for a customizing which tests must run and which must be skipped. Otherwise, if we would have the settings in a Cypress enviroment side too, it can cause a confusion for a programmer what mode is tested and which test we want to run. Thus, we are able to determine the mode after page is loaded.

Reply:
Still not really clear, but I don't wanna block it, we can surely do that later.